libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustRoomFactory.kt

```diff
@@ -110,30 +110,30 @@ class RustRoomFactory(
                 Timber.d("Room factory is destroyed, returning null for $roomId")
                 return@withContext null
             }
-            val roomReferences = awaitRoomReferences(roomId) ?: return@withContext null
-
-            if (roomReferences.room.membership() == Membership.JOINED) {
-                val baseRoom = getBaseRoom(roomReferences) ?: return@withContext null
+            val roomListItem = awaitRoomListItem(roomId) ?: return@withContext null
 
+            if (roomListItem.membership() == Membership.JOINED) {
                 // Init the live timeline in the SDK from the RoomListItem
-                if (!roomReferences.roomListItem.isTimelineInitialized()) {
-                    roomReferences.roomListItem.initTimeline(eventFilters, "LIVE")
+                val sdkRoom = if (!roomListItem.isTimelineInitialized()) {
```

Comment:
`fullRoomWithTimeline` is already checking if it's initialized, so I think we can safely remove the check here